components/Trailer.vue

```diff
@@ -1,22 +1,288 @@
-      width="100%"
-    />
+  <div id="wrapper" class="rounded-lg pa-2">
+    <div class="play-backdrop" @click="closeVideo"></div>
+    <div class="play-button">
+      <svg class="play-circles" viewBox="0 0 152 152">
+        <circle
+          class="play-circle-01"
+          fill="none"
+          stroke="#fff"
+          stroke-width="3"
+          stroke-dasharray="343 343"
+          cx="76"
+          cy="76"
+          r="72.7"
+        />
+        <circle
+          class="play-circle-02"
+          fill="none"
+          stroke="#fff"
+          stroke-width="3"
+          stroke-dasharray="309 309"
+          cx="76"
+          cy="76"
+          r="65.5"
+        />
+      </svg>
+      <div class="play-perspective">
+        <button class="play-close" @click="closeVideo"></button>
+        <div
+          class="play-triangle"
+          @click="playVideo"
+          @mouseover="rotateCircles"
+          @mouseleave="rotateCirclesRev"
+        >
+          <div class="play-video">
+            <youtube
+              ref="youtube"
+              :video-id="videoId"
+              :player-vars="playerVars"
+              max-width="600px"
+              width="90%"
+              height="400px"
+            />
+          </div>
+        </div>
+      </div>
+    </div>
   </div>
 </template>
 
 <script>
 export default {
   data: () => {
     return {
+      openTL: null,
+      rotateTL: null,
       videoId: 'b9pBQMns4lw',
       playerVars: { autoplay: 1 },
     }
   },
+  methods: {
+    positionCircles() {
+      this.$gsap.set('.play-circle-01', {
+        rotation: 90,
+        transformOrigin: 'center',
+      })
+
+      this.$gsap.set('.play-circle-02', {
+        rotation: -90,
+        transformOrigin: 'center',
+      })
+
+      this.$gsap.set('.play-perspective', {
+        xPercent: 6.5,
+        scale: 0.175,
+        transformOrigin: 'center',
+        perspective: 1,
+      })
+
+      this.$gsap.set('.play-video', {
+        visibility: 'hidden',
+        opacity: 0,
+      })
+
+      this.$gsap.set('.play-triangle', {
+        transformOrigin: 'left center',
+        transformStyle: 'preserve-3d',
+        rotationY: 10,
+        scaleX: 2,
+      })
+    },
+    configRotate() {
+      this.rotateTL = this.$gsap
+        .timeline({ paused: true })
+        .to(
+          '.play-circle-01',
+          {
+            duration: 0.7,
+            opacity: 0.1,
+            rotation: '+=360',
+            strokeDasharray: '456 456',
+            ease: 'power1.inOut',
+          },
+          0
+        )
+        .to(
+          '.play-circle-02',
+          {
+            duration: 0.7,
+            opacity: 0.1,
+            rotation: '-=360',
+            strokeDasharray: '411 411',
+            ease: 'power1.inOut',
+          },
+          0
+        )
+    },
+    configOpen() {
+      this.openTL = this.$gsap
+        .timeline({ paused: true })
+        .to(
+          '.play-backdrop',
```

Comment:
I think you can use ids instead over here as you will switch class to tailwind classes